zerver/views/__init__.py

```diff
@@ -299,16 +299,13 @@ def accounts_accept_terms(request):
 
 def create_preregistration_user(email, request, realm_creation=False):
     # type: (text_type, HttpRequest, bool) -> HttpResponse
-    domain = request.session.get("domain")
-    realm = get_realm(domain)
-    if completely_open(realm):
-        # Clear the "domain" from the session object; it's no longer needed
-        request.session["domain"] = None
-
+    domain = request.session.pop('domain', None)
+    if domain is not None:
+        # domain was set in accounts_home_with_domain.
         # The user is trying to sign up for a completely open realm,
         # so create them a PreregistrationUser for that realm
         return PreregistrationUser.objects.create(email=email,
-                                                  realm=realm,
+                                                  realm=get_realm(domain),
```

Comment:
Why do we need to be storing anything in the session?  It seems like we might be able to just pass this flag as an argument to `accounts_home` (so it goes into PreregistrationUser there) as a further cleanup...